new elastic listener role, api version control and custom container name

ld-at-work opened the pull request:
This change adds:

* a new role just for elastic listener
* The ability to set / change the api version number
* Custom Container name, the name is equal to the elk namespace, which should make things easier to find and manage.

ld-at-work committed:
Update eventhub.tf

ld-at-work committed:
Update locals.tf

ld-at-work committed:
Update variables.tf

ld-at-work committed:
Update elastic-azure-log-intergration.tf

ld-at-work committed:
Update elastic-azure-log-intergration.tf

ld-at-work commented on elastic-azure-log-intergration.tf:
```diff
@@ -39,6 +39,7 @@ resource "null_resource" "create_package_policy" {
             "parse_message": true,
             "preserve_original_event": false,
             "data_stream.dataset": "azure.eventhub",
+            "storage_account_container": "${var.elk_namespace}",
```
Adding this because I wanted to make the account container a bit more useful, it currently defaults to a very silly name

ld-at-work commented on eventhub.tf:
```diff
@@ -10,9 +10,20 @@ resource "azurerm_eventhub_authorization_rule" "root_manage" {
   name                = "RootManageSharedAccessKey"
   namespace_name      = var.event_hub_namespace
   resource_group_name = var.resource_group_name
-  eventhub_name       = azurerm_eventhub.logging.name  # Replace with your Event Hub name
```
wasnt sure why this had a #replace with your event hub name when the variable came from the resource...

ld-at-work commented on elastic-azure-log-intergration.tf:
```diff
@@ -24,7 +24,7 @@ resource "null_resource" "create_package_policy" {
   "package": {
     "name": "azure",
-    "version": "1.7.0"
+    "version": "${var.elastic_version}"
```
makes this easier to change at the code base.

gyro2009 commented on eventhub.tf:
```diff
@@ -10,9 +10,20 @@ resource "azurerm_eventhub_authorization_rule" "root_manage" {
\ No newline at end of file
+}
+
+resource "azurerm_eventhub_authorization_rule" "logging_listen" {
```
Is there any need for the `root_manage` rule now we have this rule in place with lesser permissions? We do not output it anywhere as `logging_listen` replaces the connection string.

ld-at-work replied:
probably not, if we output the eventhub details then people can add their own wants to the list? maybe?

richl-ukho commented:
Other than removing root_manage, it LGTM

ld-at-work committed:
Remove RootManage from hub

ld-at-work committed:
outputs for extending own permissions

ld-at-work committed:
maybe need a send rule?

ld-at-work committed:
Update locals.tf

ld-at-work committed:
more outputs

ld-at-work commented:
> Other than removing root_manage, it LGTM

I have removed the root_manage, but added a default logging send with the logging listen, and added both primary connection strings to an output, I hope....

ld-at-work commented:
@gyro2009 @richl-ukho if you could give this a final once over? or two or three.

ld-at-work committed:
Removes control of listener account from module

ld-at-work committed:
Integrations controlled my automation

ld-at-work committed:
Removing unncessary variables

ld-at-work committed:
removing variables

ld-at-work committed:
Update locals.tf

ld-at-work committed:
Update output.tf

ld-at-work committed:
Update eventhub.tf

ld-at-work committed:
Adds environment on the end

ld-at-work committed:
Update eventhub.tf
